src/formatter.js

```diff
@@ -4,29 +4,36 @@ var d3 = require('d3');
 var format = {};
 
 format.formatNumber = function (value, unit) {
-  if (!_.isNumber(value)) {
+  if (!_.isNumber(value) || value === 0) {
     return value;
   }
 
   var format = d3.format('.2s');
+  var p = 0;
+  var abs_v = Math.abs(value);
 
-  if (value < 1000) {
-    var v = (value).toFixed(2);
-    // v ends with .00
-    if (v.match('.00' + '$')) {
-      v = v.replace('.00', '');
-    }
-    return v;
+  if (value > 1000) {
+    value = format(value) + (unit ? ' ' + unit : '');
+    return value;
   }
 
-  value = format(value) + (unit ? ' ' + unit : '');
+  if (abs_v > 100) {
+    p = 0;
+  } else if (abs_v > 10) {
+    p = 1;
+  } else if (abs_v > 1) {
+    p = 2;
+  } else if (abs_v > 0) {
+    p = 3;
```

Comment:
if abs_v = 0.0001 then p = 3 so 0.0001.toFixed(3) is 0.000 which does not make any sense.

I'd use a logarigthm here (when under 1) to see how many of them I want, like

Math.ceil(Math.abs(Math.log(0.0001)/Math.log(10))) -> 4



toFixed( 4 + 2)